src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1411,6 +1411,35 @@ private boolean validateServerName(String nameInCert) throws CertificateExceptio
                     logger.finer(logContext + " Failed to parse the name from the certificate or name is empty.");
                 return false;
             }
+            
+            // Respect wildcard
+            if (nameInCert.contains("*")) {
+                // We do not allow wildcards to exist past the first period.
+                if (nameInCert.indexOf("*") > nameInCert.indexOf(".")) {
+                    return false;
+                }
+                String certBeforeWildcard = nameInCert.substring(0, nameInCert.indexOf("*"));
+                int firstPeriodAfterWildcard = nameInCert.indexOf(".", nameInCert.indexOf("*"));
```

Comment:
Not sure how relevant this is realistically but we should retain the position of "*" and "." since it seems we're using it a fair amount, but traversing the string everytime with indexOf().

Another thing is we check `nameInCert.contains("*")` at the beginning, but then immediately traverse the string to search for "*" again to see if it's before ".". This logic can probably be cleaned up to traverse once as `indexOf()` returns -1 if the character isn't found.

Reply:
Good suggestion. I've reflected this in my newest commit.